Autopilot/Src/altimeter.cpp

```diff
@@ -0,0 +1,136 @@
+#include "altimeter.hpp"
+
+#include "stm32f7xx_hal.h"
+#include "stm32f7xx_hal_i2c.h"
+
+#include <algorithm>
+
+/***********************************************************************************************************************
+ * Definitions
+ **********************************************************************************************************************/
+
+#define I2C_BUS 1
+#define LEFT_SHIFTED_ALTIMETER_SLAVE_ADDRESS 0xC0 // actual slave address is 0x60. But for whatever reason, the cubeMX drivers don't left shift the address so I have to do that myself
+#define ALT_CTRL_REG1 0x26
+#define ALT_OUT_P_MSB_REG 0x01
+#define STANDBY_MODE_8_OS 0x98
+#define ACTIVE_MODE_8_OS 0x99
+
+#define MEM_SIZE_1_B 1
+
+/***********************************************************************************************************************
+ * Static variables
+ **********************************************************************************************************************/
+
+static bool dataIsNew;
+
+/***********************************************************************************************************************
+ * HAL SPI handle
+ **********************************************************************************************************************/
+
+extern I2C_HandleTypeDef hi2c4;
+
+/***********************************************************************************************************************
+ * Public methods
+ ***********************************************************************:o**********************************************/
+
+Altimeter& MPL3115A2::getInstance()
+{
+    static MPL3115A2 singleton;
+    return singleton;
+}
+
+void MPL3115A2::Begin_Measuring(void)
+{
+    HAL_I2C_Mem_Read(&hi2c4, LEFT_SHIFTED_ALTIMETER_SLAVE_ADDRESS, ALT_OUT_P_MSB_REG, MEM_SIZE_1_B, rawAltimeter.byte, 1, 10);
+}
+
+static float dataLog[100];
```

Comment:
declare and define file-global variables at top pls

Reply:
forgot to get rid of this. It was just for some local tests I ran.